pkg/monitoring/persistent_volumes_collector.go

```diff
@@ -0,0 +1,119 @@
+package monitoring
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/prometheus/client_golang/prometheus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"kubevirt.io/containerized-data-importer/pkg/common"
+)
+
+var (
+	log = logf.Log.WithName("persistent-volume-info-collector")
+
+	metrics = map[MetricsKey]*metric{
+		PersistentVolumeInfo: newMetric(
+			MetricOptsList[PersistentVolumeInfo].Name, MetricOptsList[PersistentVolumeInfo].Help,
+			[]string{"pvc_name", "namespace", "pv_name", "volumemode", "volumeattributes"},
+			prometheusTypeStringToValueType(MetricOptsList[PersistentVolumeInfo].Type),
+		),
+	}
+)
+
+// PersistentVolumeInfoCollector collects information about PersistentVolumes managed by CDI
+type PersistentVolumeInfoCollector struct {
+	client *kubernetes.Clientset
+}
+
+// SetupPersistentVolumesCollector registers PersistentVolumeInfoCollector with Prometheus
+func SetupPersistentVolumesCollector(client *kubernetes.Clientset) PersistentVolumeInfoCollector {
+	c := PersistentVolumeInfoCollector{
+		client: client,
+	}
+	prometheus.MustRegister(c)
+	return c
+}
+
+// Collect implements prometheus.Collector interface
+func (c PersistentVolumeInfoCollector) Collect(ch chan<- prometheus.Metric) {
+	c.collectPersistentVolumeInfo(ch)
+}
+
+// Describe implements prometheus.Collector interface
+func (c PersistentVolumeInfoCollector) Describe(ch chan<- *prometheus.Desc) {
+	for _, metric := range metrics {
+		ch <- metric.desc
+	}
+}
+
+func (c PersistentVolumeInfoCollector) collectPersistentVolumeInfo(ch chan<- prometheus.Metric) {
+	pvcs, err := c.getPVCsManagedByCDI()
+	if err != nil {
+		log.Error(err, "Failed to get PVCs managed by CDI")
+		return
+	}
+
+	for _, pvc := range pvcs {
+		c.createVolumeInfoMetric(ch, pvc)
+	}
+}
+
+func (c PersistentVolumeInfoCollector) getPVCsManagedByCDI() ([]v1.PersistentVolumeClaim, error) {
+	pvcs, err := c.client.CoreV1().PersistentVolumeClaims("").List(
+		context.TODO(),
+		metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s", common.AppKubernetesManagedByLabel, "cdi-controller"),
+		},
+	)
+
+	if err != nil {
+		return nil, err
```

Comment:
I don't think we want to return an error when the returned error is 'is not found' Which just means there are no PVCs yet. That is a valid state and not a real error.